### PR TITLE
[WFMP-208]: dev mode deployment runtime name is not configurable.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding a new 'name' parameter with the same property as for the package mojo.

Jira: https://issues.redhat.com/browse/WFMP-208

### CODE REVIEW

File: plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java [hunk 1]
```diff
@@ -837,7 +845,8 @@ private Deployment getDeploymentContent() {
         if (remote) {
             return Deployment.of(Path.of(project.getBuild().getDirectory()).resolve(filename));
         } else {
-            return Deployment.of(resolveWarDir()).setRuntimeName(filename);
+            String runtimeName = this.name == null || this.name.trim().isBlank() ? filename : this.name;
```
Comment (jamezp) [inline plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java:848]: There is no need to use `trim()` with `isBlank()`

File: plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java [hunk 2]
```diff
@@ -342,6 +342,14 @@ public class DevMojo extends AbstractServerStartMojo {
     @Parameter(alias = "channels")
     private List<ChannelConfiguration> channels;
 
+    /**
+     * Specifies the name used for the deployment.
+     *
+     * When the deployment is copied to the server, it is renamed with this name.
+     */
+    @Parameter(property = PropertyNames.DEPLOYMENT_NAME)
```
Comment (jamezp) [inline plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java:350]: I'm on my phone so it's hard to see, but if the run goal has this field, we should remove it from there. Or just move this field into the `DevMojo`.

**Review by jamezp — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java (hunk 2 above)
Comment (ehsavoie) [reply plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java:350]: yes it has but not the StartMojo also the RunMojo has runtimeName which is deperecated in the PackageServerMojo

**Review by jamezp — APPROVED** [review]

**Review by jamezp — CHANGES_REQUESTED** [review]
This looks like it has formatting issues.

File: plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java (hunk 2 above)
Comment (ehsavoie) [reply plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java:350]: @jamezp I don't see the point of adding this field to the 'parent' class AbstractServerStartMojo and it doesn't make sense in the context of StartMojo. So I don't think we should 'move' it out :)

**Comment by jamezp** [comment]
Replaces by #350 to get this change into the next release.

**Pull request closed without merging**